src/bin/zmcontactbackup

```diff
@@ -0,0 +1,565 @@
+# the terms of the GNU General Public License as published by the Free Software Foundation,
+# version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this program.
+# If not, see <https://www.gnu.org/licenses/>.
+# ***** END LICENSE BLOCK *****
+
+=head1 NAME
+
+zmcontactbackup - start/stop/schedule contact backup on mailbox server(s)
+
+=head1 SYNOPSIS
+
+zmcontactbackup <-h|-m|-r|-p|-q|-d|-f|[-a backup_schedule]> [mailbox_server...]
+
+       -r|--start          : start contact backup
+       -p|--stop           : stop contact backup
+       -h|--help           : print a brief help message
+       -m|--man            : print full man message
+       -q|--query          : print existing schedule
+       -a|--append         : append contact backup schedule
+       -f|--flush          : flush all contact backup schedules
+       -d|--default        : (re)set to the default (3am daily) backup schedule
+       backup_schedule     : crontab style time specifier, QUOTED.  See crontab(5)
+                Fields are:
+                        minute         0-59
+                        hour           0-23
+                        day of month   1-31
+                        month          1-12
+                        day of week    0-7 (0 or 7 where 0 is Sunday, or use names like sun, mon, etc...)
+                Example: Everyday at 1am  -  "0 1 * * *"
+        mailbox_server : space separated mailbox server names
+
+=head1 EXAMPLES
+
+Examples :
+1)  # append contact backup schedule to start everyday at 1 am on all mailbox servers
+    $ zmcontactbackup -a "0 1 * * *"
+
+2)  # append contact backup schedule to start everyday at 1 am on "test.server.com" only.
+    $ zmcontactbackup -a "0 1 * * *" test.server.com
+
+3)  # start contact backup on all mailbox servers
+    $ zmcontactbackup -r
+
+4)  # start contact backup on mailbox servers test.server.com and test2.server.com
+    $ zmcontactbackup -r test.server.com test2.server.com
+
+5)  # stop contact backup on all mailbox servers if running.
+    $ zmcontactbackup -p
+
+6)  # stop contact backup on test.server.com, if running. do not disturb running contact backup on other mailbox servers.
+    $ zmcontactbackup -p test.server.com
+
+=head1 DESCRIPTION
+
+This utility is used to schedule contact backup for
+specified mailbox server(s).
+It can be used to start/stop contact backup immediately.
+It uses the ZCS Admin SOAP APIs.
+
+Specifically, this utility was created to meet the following
+requirements:
+1) start contact backup thread
+2) stop contact backup thread
+3) schecule contact backup to run on specific time
+4) contact backup should run on all or specific mailbox servers.
```

Comment:
Note: the "all" case is handled by:
1. installing the cron job on each mailbox server
2. specifying all mailbox servers on the command line